plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/scopeanalysis/AbstractScopeAnalyzerVisitor.java

```diff
@@ -327,8 +328,9 @@ protected void addToNamesToIgnore(SimpleNode node, boolean finishClassScope, boo
                 continue;
             }
             IToken tok = single.tok;
-            String rep = tok.getRepresentation();
-            if (rep.equals(token.getRepresentation())) {
+            List<String> repParts = StringUtils.split(tok.getRepresentation(), '.');
+
+            if (repParts.contains(token.getRepresentation())) {
```

Comment:
I think you can't blindly check like that, you need to check if the first part is actually the current class name.

Something as:

```
            String firstPart = FullRepIterable.getFirstPart(tok.getRepresentation());
            if (firstPart.equals(token.getRepresentation())) {
```

Otherwise if you had something as:

`m.C` in that case would end up validating when it should give an error.

Reply:
Yup. When I followed this approach, I thought that there could be cases where the first part of the attribute is not the class itself. But looking now I don't think it's possible....